pydantic/_internal/_generate_schema.py

```diff
@@ -996,14 +996,22 @@ def json_schema_update_func(schema: CoreSchemaOrField, handler: GetJsonSchemaHan
 
         # apply alias generator
         alias_generator = self._config_wrapper.alias_generator
-        if alias_generator and (field_info.alias_priority is None or field_info.alias_priority <= 1):
-            alias = alias_generator(name)
-            if not isinstance(alias, str):
-                raise TypeError(f'alias_generator {alias_generator} must return str, not {alias.__class__}')
-            field_info.alias = alias
-            field_info.validation_alias = alias
-            field_info.serialization_alias = alias
-            field_info.alias_priority = 1
+        if alias_generator:
+            if field_info.alias_priority is None or field_info.alias_priority <= 1:
+                alias = alias_generator(name)
+                if not isinstance(alias, str):
+                    raise TypeError(f'alias_generator {alias_generator} must return str, not {alias.__class__}')
+                field_info.alias = alias
+                field_info.validation_alias = alias
+                field_info.serialization_alias = alias
```

Comment:
These 6 lines seem exactly copied between the two if branches. I think it would be better to not duplicate them, and just include the `field_info.alias_priority = 1` line when `field_info.alias is not None`. So this would end up looking more like:

```python
            if field_info.alias_priority is None or field_info.alias_priority <= 1 or field_info.alias is None:
                alias = alias_generator(name)
                if not isinstance(alias, str):
                    raise TypeError(f'alias_generator {alias_generator} must return str, not {alias.__class__}')
                field_info.alias = alias
                field_info.validation_alias = alias
                field_info.serialization_alias = alias
                if field_info.alias is not None:
                    field_info.alias_priority = 1
```